### PR TITLE
Device Port Error changes splitup - 3

### PR DESCRIPTION
**Pull request by kalyan-nidumolu**
1) Changed ParseError / ParseErrorTime in NetworkPortConfig to ErrorAndTime
2)Cleaned up Error messages
3) Added DevicePortConfig.SetPortErrorByIfname

### CODE REVIEW

File: pkg/pillar/cmd/zedagent/parseconfig.go [hunk 1]
```diff
@@ -741,20 +739,22 @@ func parseOneSystemAdapterConfig(getconfigCtx *getconfigContext,
...
 		} else {
 			net := networkXObject.(types.NetworkXObjectConfig)
 			port.NetworkUUID = net.UUID
 			network = &net
 			if network.HasError() {
-				errStr := fmt.Sprintf("parseSystemAdapterConfig: Port %s Network error: %v",
-					port.IfName, network.Error)
-				port.ParseError = errStr
-				port.ParseErrorTime = network.ErrorTime
+				errStr := fmt.Sprintf("Port %s configured with Error Network. "+
```
Comment (eriknordmark) [inline pkg/pillar/cmd/zedagent/parseconfig.go:753]: Thanks for making the error strings more usable for the admin.

But "with Error Network" doesn't read well. Should it say "with a network which has a (posted?) error. Network UUID: %s, Network Error: %s"?

File: pkg/pillar/cmd/zedagent/parseconfig.go [hunk 2]
```diff
@@ -775,39 +775,37 @@ func parseOneSystemAdapterConfig(getconfigCtx *getconfigContext,
 		switch port.Dhcp {
 		case types.DT_STATIC:
 			if port.AddrSubnet == "" {
-				errStr := fmt.Sprintf("parseSystemAdapterConfig: Port %s DT_STATIC but missing "+
-					"subnet address in %+v; ignored", port.IfName, port)
-				log.Error(errStr)
-				port.ParseError = errStr
-				port.ParseErrorTime = time.Now()
+				errStr := fmt.Sprintf("Port %s Configured as DT_STATIC but "+
```
Comment (eriknordmark) [inline pkg/pillar/cmd/zedagent/parseconfig.go:778]: DT_STATIC in the code. In the UI we might be calling this just static. Maybe saying "configured as having static IP but missing a subnet address".

File: pkg/pillar/cmd/zedagent/parseconfig.go [hunk 3]
```diff
@@ -775,39 +775,37 @@ func parseOneSystemAdapterConfig(getconfigCtx *getconfigContext,
...
+					"missing subnet address. Port: %+v.", port.IfName, port)
```
Comment (eriknordmark) [inline pkg/pillar/cmd/zedagent/parseconfig.go:779]: %v for the whole port isn't useful to the admin. Maybe printing the sysAdapter.Addr in addition to the IfName would be helpful.

File: pkg/pillar/cmd/zedagent/parseconfig.go [hunk 4]
```diff
@@ -775,39 +775,37 @@ func parseOneSystemAdapterConfig(getconfigCtx *getconfigContext,
...
 		case types.DT_NONE:
 			if isMgmt {
-				errStr := fmt.Sprintf("parseSystemAdapterConfig: Port %s: isMgmt with DT_NONE not supported",
-					port.IfName)
-				log.Error(errStr)
-				port.ParseError = errStr
-				port.ParseErrorTime = time.Now()
+				errStr := fmt.Sprintf("Port %s configured as Management port "+
+					"with DHCP type DT_NONE. Only Client and Static are "+
+					"DHCP modes for management ports", port.IfName)
```
Comment (eriknordmark) [inline pkg/pillar/cmd/zedagent/parseconfig.go:789]: I don't know what DT_NONE is called in the UI.Maybe say
"with an unsupported DHCP type %d. Client and static are the only allowed DHCP modes for management ports".

File: pkg/pillar/cmd/zedagent/parseconfig.go (hunk 1 above)
Comment (kalyan-nidumolu) [reply pkg/pillar/cmd/zedagent/parseconfig.go:753]: Done..	
	errStr := fmt.Sprintf("Port %s configured with a network "+
					"(UUID: %s) which has an error (%s).",
					port.IfName, port.NetworkUUID, network.Error)

File: pkg/pillar/cmd/zedagent/parseconfig.go (hunk 3 above)
Comment (kalyan-nidumolu) [reply pkg/pillar/cmd/zedagent/parseconfig.go:779]: Done.. Changed it to:

				errStr := fmt.Sprintf("Port %s Configured as DT_STATIC but "+
					"missing subnet address. SysAdapter - Name: %s, Addr:%s",
					port.IfName, sysAdapter.Name, sysAdapter.Addr)

File: pkg/pillar/cmd/zedagent/parseconfig.go (hunk 4 above)
Comment (kalyan-nidumolu) [reply pkg/pillar/cmd/zedagent/parseconfig.go:789]: Done.. Changed it to:

	errStr := fmt.Sprintf("Port %s configured as Management port "+
					"with an unsupported DHCP type %d. Client and static are "+
					"the only allowed DHCP modes for management ports.",
					port.IfName, types.DT_NONE)

**Pull request merged**